### ISSUE TITLE
Use a build tool to automatically make a new release

### ISSUE THREAD

**Issue by N-Wouda** [ndvanforeest/queueing_book#531]
Merging our code in is good fun, but does require you to manually recompile the book every time we edit stuff. We can do better, by using e.g. a build tool like [Travis CI](https://travis-ci.com/) to do that for us. After building the document, it can even automatically release the book as a [Github release](https://github.com/ndvanforeest/queueing_book/releases). The static url to the latest version of the book is then always:

```https://github.com/ndvanforeest/queueing_book/releases/latest```

And any changes automatically get published there once the code is merged into `master`.

Travis is free for academic use, and so far it's worked very well for me in my own projects.

**Comment by ndvanforeest** [comment]
I considered this, but... can travis also run pythontex with python3? And how about latex with all the extra packages I use? If so, let me know. It would indeed save quite some time.

**Comment by N-Wouda** [comment]
I suspect so: you get full access to a container (generally Ubuntu, but you can decide on a different OS), which you can do with as you please. I've so far had no trouble installing any package I needed there. If it's available for the OS, you can install it yourself.

In my most recent project I run py3.7, a host of packages, and a Postgres + PostGIS database, all on Travis to run my tests. Took a little time to get it configured nicely, but I never ran into something that wouldn't work.

**Comment by ndvanforeest** [comment]
That sounds good. Perhaps we should discuss this after class, and see what to do.

**Comment by N-Wouda** [comment]
@ndvanforeest if you mean tomorrow, I'd suggest after the midterm instead. I'm not in class tomorrow due to a meeting for work.

**Comment by ndvanforeest** [comment]
I noticed that the prs of my students sometimes contain errors so that the latex files don’t compile. I guess it is best that I keep on handling such things. Unless Travis also learns how deal with latex errors and queueing. 😀

**Comment by N-Wouda** [comment]
> I noticed that the prs of my students sometimes contain errors so that the latex files don’t compile. I guess it is best that I keep on handling such things. Unless Travis also learns how deal with latex errors and queueing. 😀

It already can - that'd fail the build (non-zero exit code).
